### ISSUE TITLE
Jumper Ezpad 6 Pro Touchscreen far off.... 

### ISSUE THREAD

**Issue by Dantsze** [onitake/gsl-firmware#148]
Hey,

so I got myself a Ezpad 6 Pro and install Linux and everything works but except... You guessed it the touchscreen. So I was looking for a tutorial on how to install the drivers for it on linux, and I found one for the Ezpad 5S but I think shouldn't be that different except for the firmware and so on.... But I'm having some difficulties to understand that tutorial and how to make it work for the Ezpad 6 Pro, so could you help making a step by step tutorial on how to install the Ezpad 6 Pro silead drivers for linux?

This is what I found:

Here’s the procedure, executed on Linuxian 16.10 beta 2 (it should work on any other recent distro, I think) ( Ok, I'm an on different but debian based, as he said it should work...

1. Install git if not yet installed and download the firmware / tools

sudo apt-get install git 
Question: after the git install do I have to use my user and email after that?

git clone https://github.com/onitake/gsl-firmware.git
Question: Shouldn't I clone the firmware git of the Jumper Ezpad 6 Pro instead?

2. Extract firmware for your original windows file (you can use my in attachment to see if it works) and copy firmware to correct folder
Question:  Can't I just use the firmware which is uploaded under the Ezpad 6 Pro git instead of extracting it from windows?

copy the SileadTouch.fw to gsl-firmware/tools and run
Question: I don't understand this one.... I mean how should I do it? Can someone explain what he means? Can someone explains this in a commando line or something?

unscramble SileadTouch.fw firmware.fw
./fwtool -c firmware.fw -m 1680 -w 1920 -h 1080 -t 2 silead_ts.fw
(Right here I would use the Jumper Ezpad Pro 6 line under readme instead of this one...)

sudo cp silead_ts.fw /lib/firmware/

3. compile and install driver
Question: How should I compile and install the driver? If I would make this far, my question should be already sloved.

go back to the folder where you cloned the firmware/tools and

git clone https://github.com/onitake/gslx680-acpi.git
cd gslx680-acpi
make
sudo cp gslx680_ts_acpi.ko /lib/modules/$(uname -r)/kernel/drivers/

4. Testing

at this point you can test the touchscreen.
In the same folder as the step above just do:

sudo insmod ./gslx680_ts_acpi.ko

You should now have a working and not calibrated touch screen

5. Calibrating touch

It’s good to have a tool for that.
Just install xinput_calibrator:

sudo apt-get install xinput_calibrator
xinput_calibrator

and touch the screen on the points.

It will output a configuration text that you should put on a file (or use mine in attachment)

6. Make everything permanent:

sudo gedit /usr/share/X11/xorg.conf.d/99-calibration.conf

and paste your config output, or mine:

Section "InputClass"
Identifier "calibration"
MatchProduct "Silead GSLx680 Touchscreen"
Option "Calibration" "5 1958 26 1545"
Option "SwapAxes" "0"
EndSection

As we’ve already copied the driver to /lib/modules/$(uname -r)/kernel/drivers/ now we just need to tell the system to load it on boot:

sudo echo 'gslx680_ts_acpi' | sudo tee -a /etc/modules
sudo depmod

7. Reboot and you’re done.

https://www.youtube.com/watch?v=puh3L6tG-Ec
Too bad, that video doesn't work...

I took the tutorial from this link: https://techtablets.com/forum/topic/ezpad-5s-dual-boot-2/#post-52678

Can someone help me and make this tutorial a little more specific, detailed and simpler?

**Comment by Dantsze** [comment]
So I made it this far:

sudo apt-get install git
git clone https://github.com/onitake/gsl-firmware.git

and then I copied the silead_ts.fw and the firmware from jumper ezpad pro6 to .gsl-firmware/tools 
and then I've tried to run unscramble didn't do anything...

tried: 
unscramble silead_ts.fw firmware.fw 
also didn't to anything..

and then I tried: 
.gsl-firmware/tools/unscramble silead_ts.fw firmware.fw 

and:
.gsl-firmware/tools/unscramble unscramble silead_ts.fw firmware.fw 
also didn't work

What I'm doing wrong? Aren't those files the files extracted from windows?
What are does files then and how do I use them?
